server/forms.py

```diff
@@ -1,45 +1,57 @@
 from flask_wtf import Form
-from wtforms import TextField, DateTimeField, BooleanField, IntegerField
-from wtforms import validators
+from wtforms import StringField, DateTimeField, BooleanField, IntegerField
+from wtforms import SelectField, TextAreaField, validators
+from wtforms.fields.html5 import EmailField
 
 import datetime as dt
 # from wtforms.ext.sqlalchemy.orm import model_form
 from .models import Assignment
-import pytz
+from server.constants import VALID_ROLES
 
+import pytz
+import csv
+import logging
 
-def strip_whitespace(input):
-    if hasattr(input, "strip"):
-        return input.strip()
+def strip_whitespace(value):
+    if value and hasattr(value, "strip"):
+        return value.strip()
     else:
-        return input
+        return value
 
 
 def convert_to_utc(date):
-    # TODO Not sure if 100 TZ aware. Unit test.
+    # TODO Not sure if 100% TZ aware. Unit test.
     return pytz.utc.localize(date)
 
+class BaseForm(Form):
+    class Meta:
+        def bind_field(self, form, unbound_field, options):
+            filters = unbound_field.kwargs.get('filters', [])
+            field_type = type(unbound_field)
+            if field_type == StringField:
+                filters.append(strip_whitespace)
+            elif field_type == DateTimeField:
+                filters.append(convert_to_utc)
+            return unbound_field.bind(form=form, filters=filters, **options)
 
-class AssignmentForm(Form):
-    display_name = TextField(u'Display Name',
-                             [validators.required()])
-    name = TextField(u'Offering', filters=[strip_whitespace],
-                     validators=[validators.required()])
+
+class AssignmentForm(BaseForm):
+    display_name = StringField(u'Display Name',
+                               validators=[validators.required()]),
+    name = StringField(u'Offering', validators=[validators.required()])
     due_date = DateTimeField(u'Due Date (Pacific Time)',
-                             filters=[convert_to_utc],
```

Comment:
Fix the string to reflect that it's also in pacific time. (commit is coming in) 

That would be nice - but I'm going to file it as a `nice to have`